Final draft!

glouppe opened the pull request:
Here we are! This is a final draft for the paper.

* [x] Second example in Nd with parameterized classifier and BO
* [ ] Final proofreading (Gilles)
* [ ] Final proofreading (Kyle)
* [x] Final proofreading (Juan)
* [x] Add Juan affiliation
* [x] Use greek letters for parameters (example 1)
* [x] Make Figure 1 nicer
* [x] Add chi square distribution next to figure 2
* [x] Use greek letters for parameters (example 2)
* [x] Update Figure 4
* [ ] Proofcheck notations

@cranmer @jgpavez Can you find time for a final read of this version? It is time to conclude :)

cranmer commented:
I love the example.

Misc comments:

- [ ]  reproducibility, would be nice to specify the R transformation, but this can also be done by printing R in the notebook and having it on zenodo.

- [ ] scatter points in x0 in fig 3 looks like they've been cut off in the pdf

- [ ] Fig 4 is beautiful, but puts emphasis on bayesian optimization instead of the method.
We could keep it, but would be nice to have a figure that shows:
True point used to generate samples, MLE from true likelihood, MLE from our approach.
It would also be nice to have contours from true likelihood and perhaps compare with contours from our approach.

- [ ] can we add some more runs for fig 2

- [ ] we don't have any results from diagnostic (AUC for p1 vs. p0*r(x))

- [ ]  would be nice to run current n-d example for fig 4 using different p1 in the denominator

jgpavez commented:
Wow, the paper is looking great now!, 
Some minor comments:
- In section 2.3 it is mentioned that tree-based methods does not work so well when training set is small, maybe it should be clarified than in cases that the training set is relatively large tree based methods work as well as MLPs (I have done all experiments on real data with BDTs), since tree based methods are largely used on the community.
- At the end of section 2.2 you mention that the method is expected to be more robust than directly using 1-s(x)/s(x) ... at this point maybe a reference or explanation of why 1-s(x)/s(x) can replace r(x) could be useful (something that also is only true if p(\theta_0) = p(\theta_1) = 1/2).

Some typos:
- Page 4, Theorem 1. --> it says 'be a random vector vector'
- Page 6, after theorem 2.10, it says 'satisfies conditions' maybe it should be 'satisfies the conditions'
- Page 6, after theorem 2.10, it says 'in case we learn' maybe it should be 'in case that we learn'

glouppe commented:
@jgpavez I think I now have addressed all your comments. Thanks!

glouppe commented:
> for reproducibility, would be nice to specify the R transformation, but this can also be done by printing R in the notebook and having it on zenodo.

I updated the notebook with this.

> scatter points in x0 in fig 3 looks like they've been cut off in the pdf

Not sure I can do anything about that... I did nothing special but using the `corner` module from @dfm :)

> Fig 4 is beautiful, but puts emphasis on bayesian optimization instead of the method.
We could keep it, but would be nice to have a figure that shows:
True point used to generate samples, MLE from true likelihood, MLE from our approach.
It would also be nice to have contours from true likelihood and perhaps compare with contours from our approach.

Agreed and fixed!

> can we add some more runs for fig 2

Done

glouppe commented:
> would it make sense to show p(s(x)) for either 1-d or 2-d example? in paper or at least notebook? Somehow that emphasizes the dimensionality reduction more explicitly.

> we don't have any results from diagnostic (AUC for p1 vs. p0*r(x))

> would be nice to run current n-d example for fig 4 using different p1 in the denominator

Could we explore all this in the notebooks? Not sure I want to make the paper longer than what it already is. What do you think?

jgpavez commented:
>> scatter points in x0 in fig 3 looks like they've been cut off in the pdf

> Not sure I can do anything about that... I did nothing special but using the corner module from @dfm :)

Maybe you can check https://stanford.edu/~mwaskom/software/seaborn/generated/seaborn.pairplot.html ?

glouppe commented:
@cranmer I updated the first example with our latest experiments in the notebook. Do you think the text is good enough like this? I wouldnt mind some help to improve it, there are certainly more to say or things to remove.

glouppe commented:
The second example is now updated as well. Comments and suggestions are welcome!

@cranmer Could you remind me what you had in mind for discussing diagnostics?

cranmer commented:
We had discussed a 3x2 table of plots with rows being
A) ROC curve of classifier or p0 vs r*p1
B) a few likelihood curves for different choices of p1 in denominator

And columns:
Poor training, good calibration
Poor calibration, good training
Good both

Needs to be a multidimensional X, but can do a 1-parameter likelihood. Eg.
Just one of the two parameters in second example.


On Thu, Mar 10, 2016 at 6:48 AM Gilles Louppe <notifications@github.com>
wrote:

> The second example is now updated as well. Comments and suggestions are
> welcome!
>
> @cranmer <https://github.com/cranmer> Could you remind me what you had in
> mind for discussing diagnostics?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cranmer/parametrized-learning/pull/12#issuecomment-194809183>
> .
>

jgpavez commented:
If it helps ... I did some experiments on ROC curve evaluation here ... https://github.com/jgpavez/DecomposingTests/blob/master/evaluating_notebook.ipynb
The evaluation is on the last three code cells

cranmer commented:
Examples look great. I'll work on some of the text around the figures. Some small thing like how we refer to "True MLE" is confusing I think. Prefer MLE from true likelihood.  Or better yet "exact" likelihood to make exact/approximate comparison more clear.

glouppe commented:
Thanks for the changes! There are certainly other changes worth doing regarding the text. Please feel free to go ahead, as your English is no doubt better than mine :) Meanwhile I'll work on the diagnostic plots.
